.jenkins/Jenkinsfile-pr-s390x

```diff
@@ -0,0 +1,234 @@
+                }
+            }
+        }
+        stage('Install yq') {
+            steps {
+                timeout(time: 5, unit: 'MINUTES') {
+                    script {
+                        lib.installYq(env.WORKSPACE, 's390x')
+                    }
+                }
+            }
+        }
+        stage('Build project') {
+            steps {
+                script {
+                    withMaven(mavenOpts: '-Djansi.force=true') {
+                        sh(script: "mvn clean install -DskipTests -Dstyle.color=always --no-transfer-progress")
+                        env.BUILD_PROJECT_STATUS = true
+                    }
+                }
+            }
+        }
+        stage('Build images') {
+            when {
+                environment name: 'TEST_ONLY', value: 'false'
+            }
+            steps {
+                script {
+                    //The following 2 steps are temporary
+                    lib.buildKeycloakAndOpa_s390x(env.WORKSPACE)
+                    lib.applys390xpatch(env.WORKSPACE)
+                    lib.buildStrimziImages()
+                    env.BUILD_DOCKER_IMAGE_STATUS = true
+                    env.DOCKER_REGISTRY = sh(script: "kubectl get service registry -n kube-system -o=jsonpath='{.spec.clusterIP}'", returnStdout: true).trim() + ":80"
+                    println("DOCKER_REGISTRY: ${env.DOCKER_REGISTRY}")
+                }
+            }
+        }
+        stage('Execute system tests') {
+            steps {
+                script {
+                    catchError {
+                        // For upgrade tests running with locally built images we need to change tag in StrimziUpgradeST.json
+                        if ("${env.TEST_PROFILE}".contains("upgrade")) {
+                            println("[INFO] Update files for upgrade procedure")
+                            sh(script: """
+                                sed -i 's#:latest#:${env.DOCKER_TAG}#g' ${env.WORKSPACE}/systemtest/src/test/resources/upgrade/StrimziUpgradeST.json ${env.WORKSPACE}/install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml
+                                sed -i 's#quay.io/strimzi/test-client:${env.DOCKER_TAG}#${env.DOCKER_REGISTRY}/strimzi/test-client:${env.DOCKER_TAG}#g' ${env.WORKSPACE}/systemtest/src/test/resources/upgrade/StrimziUpgradeST.json
+                                sed -i 's#quay.io/strimzi/#${env.DOCKER_REGISTRY}/strimzi/#g' ${env.WORKSPACE}/install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml
+                                sed -i 's#/opt/${env.DOCKER_REGISTRY}#/opt#g' ${env.WORKSPACE}/install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml
+                            """)
+                            sh(script: "cat ${env.WORKSPACE}/systemtest/src/test/resources/upgrade/StrimziUpgradeST.json")
+                            sh(script: "cat ${env.WORKSPACE}/install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml")
```

Comment:
I think you can put this into a method and call it in this and in x86 pipeline.

Reply:
Done.